### PR TITLE
Add Unobstructed Lane Change

### PR DESCRIPTION
**Pull request by saina-ramyar**
# PR Details
## Description

Adds the unobstructed lane-change capability to CARMA. This feature is based on the CARMA new routing feature and generates trajectory directly using the lanelet's information. Spline library is used to ensure the continuity of the generated trajectory. 

## Related Issue
#710 

## Motivation and Context
This feature allows the vehicle to change the lane when requested by the routing plugin.
## How Has This Been Tested?
This code has beem unit tested

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: unobstructed_lanechange/test/test_unobstructed_lanechange_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+TEST(UnobstructedLaneChangePluginTest, testCreateLaneChangeTrajectory1)
+{
+    std::vector<double> start_point = {0.0, 0.0};
+    std::vector<double> end_point = {10.0, 3.0};
+    unobstructed_lanechange::UnobstructedLaneChangePlugin lc;
+    std::vector<cav_msgs::TrajectoryPlanPoint> res = lc.create_lanechange_trajectory(start_point, end_point);
+    EXPECT_EQ(60, res.size());
+    EXPECT_NEAR(5.0, res[30].x, 0.01);
+    EXPECT_NEAR(1.5, res[30].y, 0.01);
```
Comment (qswawrq) [inline unobstructed_lanechange/test/test_unobstructed_lanechange_plugin.cpp:31]: I think we also need to check points in between.

File: unobstructed_lanechange/include/spline.h [hunk 2]
```diff
@@ -0,0 +1,404 @@
+/*
+ * spline.h
```
Comment (qswawrq) [inline unobstructed_lanechange/include/spline.h:2]: Not sure if you have another way to deal with it. But generally we do not want to include third party file in our repo.

File: unobstructed_lanechange/src/unobstructed_lanechange.cpp [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        return lc_trajectory;
+
+    }
+
+    std::vector<cav_msgs::TrajectoryPlanPoint> UnobstructedLaneChangePlugin::post_process_traj_points(std::vector<cav_msgs::TrajectoryPlanPoint> trajectory)
+    {
+        uint64_t current_nsec = ros::Time::now().toNSec();
+        for(int i = 0; i < trajectory.size(); ++i)
+        {
+            trajectory[i].controller_plugin_name = "mpc_follower";
```
Comment (qswawrq) [inline unobstructed_lanechange/src/unobstructed_lanechange.cpp:176]: Try to make it configurable.

**Comment by kjrush** [comment]
Closing, as this PR was updated for the new waypoint generation behaviors and integration tested under #1042. This code is merged in via that PR.

**Pull request closed without merging**